sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -197,16 +198,18 @@ class AioHttpStreamDownloadGenerator(AsyncIterator):
 
     :param pipeline: The pipeline object
     :param response: The client response object.
-    :param block_size: block size of data sent over connection.
-    :type block_size: int
+    :keyword bool decompress: If True which is default, will attempt to decode the body based
+        on the ‘content-encoding’ header.
     """
-    def __init__(self, pipeline: Pipeline, response: AsyncHttpResponse) -> None:
+    def __init__(self, pipeline: Pipeline, response: AsyncHttpResponse, **kwargs) -> None:
         self.pipeline = pipeline
         self.request = response.request
         self.response = response
         self.block_size = response.block_size
+        self._decompress = kwargs.pop("decompress", True)
```

Comment:
I suspect that we can use the status code (206 - Partial content) + the Content-Range header to determine if we can decompress by default. And have that as the default (e.g. the user didn't provide the `decompress` parameter). Did you/we consider that?